### PR TITLE
Restore Global.reporter()Reporter

### PR DESCRIPTION
**Pull request by dwijnand**
It's an important binary API:

    [scala-rewrites] [error] java.lang.NoSuchMethodError: scala.tools.nsc.Global.reporter()Lscala/tools/nsc/reporters/Reporter;
    [scala-rewrites] [error] 	at scala.meta.internal.semanticdb.scalac.SemanticdbPlugin.init(SemanticdbPlugin.scala:25)
    [scala-rewrites] [error] 	at scala.tools.nsc.plugins.Plugins.$anonfun$loadPlugins$9(Plugins.scala:162)

### CODE REVIEW

**Review by lrytz — APPROVED** [review]

**Comment by som-snytt** [comment]
I need to make a sad face somewhere, and this is as good a place as any. 😞 

I'm too lazy to find the PR where Lukas commented that it was a fool's errand to change this API to use `internal.Reporter`.

**Comment by SethTisue** [comment]
labeling with "internal" since the removal was never released

**Comment by som-snytt** [comment]
I think we need a new label `internecine`.

**Comment by dwijnand** [comment]
To answer my own question "how did we get away with breaking this in 2.13?", the answer is "because there are no binary guarantees for scala-compiler" and I'm defiantly overriding that in scala-rewrites: https://github.com/scala/scala-rewrites/blob/aa3f71bfc16c342f14566db0a730a0516acafc7e/project/ScalaNightlyPlugin.scala#L26.

In the community build both scala-compiler and scala-rewrites are compiled so if they compile together, why is it throwing a NoSuchMethodError?  I think the cause is what this warning message is getting at:

> [scala-rewrites] [warn] **** Missing dependency: the library org.scalameta#semanticdb-scalac is not provided (in space "scala.scala_rewrites") by any project in this configuration file.
> [scala-rewrites] [warn] The library (and possibly some of its dependencies) will be retrieved from the external repositories.
> [scala-rewrites] [warn] In order to control which version is used, you may want to add this dependency to the dbuild configuration file.

Which is actually also documented at the top of the CB's proj/scala-rewrites.conf:

> // This project is unusual in that we build it using only binary
> // dependencies, since the community build's scalafix version is too
> // old and we can't upgrade it because it has too many dependencies.
> // (And scala-rewrites is 2.12-only because it lives in
> // sbt-plugin-land.)
> 
> // This is trickier-than-normal because semanticdb-scalac and
> // scalafix-testkit are CrossVersion.full, so we have to override that
> // to use older binaries, since binaries don't exist for Scala nightlies.
> //
> // (If those older binaries don't work, then we'll have to do
> // something else, but in a way that's good because we can notify Olaf
> // upstream that we changed something in Scala that's going to require
> // changes to Scalafix.)

TL;DR this change wasn't as necessary as I'd thought.

**Comment by dwijnand** [comment]
Undoing this: #9300.
